src/editor/editor.js

```diff
@@ -189,6 +193,42 @@ export default class Editor {
 
 mix( Editor, EmitterMixin );
 
+// @param {module:utils/config~Config} config
+// @param {Object} builtInConfig
+// @param {Object.<String,*>} builtInConfig.config
+// @param {Array.<module:core/plugin~Plugin>} availablePlugins
+function extendEditorConfig( config, builtInConfig, availablePlugins ) {
```

Comment:
Ok. When you will be ready, put your idea to https://github.com/ckeditor/ckeditor5-core/pull/72.